test.js

```diff
@@ -0,0 +1,181 @@
+'use strict'
+
+var test = require('tape').test
+var aedes = require('aedes')
+var mqtt = require('mqtt')
+var mqttPacket = require('mqtt-packet')
+var net = require('net')
+var proxyProtocol = require('proxy-protocol-js')
+var createServer = require('./lib/server-factory')
+
+// This test suite will be really effective after updating aedes and protocol-decoder module,
+// to retrieve conn details in the client @preConnect
+
```

Comment:
Like mentionned above, these tests are rather dummy, at least we ensure that the mqtt packet is correctly parsed even after stream manipulation, but that would be better to also check `client.connDetails` after aedes has been updated.